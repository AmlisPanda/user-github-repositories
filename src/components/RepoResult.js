import React from 'react';
import moment from 'moment';

export class RepoResult extends React.Component {

    render() {

        const updAt = moment(this.props.repo.updated_at).utc();
        return (
            <article>
                <h3><a href={this.props.repo.svn_url} target="_blank">{this.props.repo.name}</a></h3>
                <p>{this.props.repo.description}</p>
                <div className="updatedInfo" title={this.props.repo.updated_at}>Updated {updAt.fromNow()}</div>
            </article>
        )
    }
}
