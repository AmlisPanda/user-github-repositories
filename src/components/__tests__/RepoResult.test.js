import React from 'react';
import renderer from 'react-test-renderer';
import { RepoResult } from '../RepoResult';

describe('RepoResult', () => {
    it('renders the component with props', () => {
        const repo = {
            updated_at: "",
            name: "Androguard",
            svn_url: "https://github.com/google/androguard",
            description: "Reverse engineering, Malware and goodware analysis of Android applications ... and more (ninja !)"
        }
        const tree = renderer.create(<RepoResult repo={repo} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

})
