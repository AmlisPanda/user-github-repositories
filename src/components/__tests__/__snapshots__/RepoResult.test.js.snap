// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoResult renders the component with props 1`] = `
<article>
  <h3>
    <a
      href="https://github.com/google/androguard"
      target="_blank"
    >
      Androguard
    </a>
  </h3>
  <p>
    Reverse engineering, Malware and goodware analysis of Android applications ... and more (ninja !)
  </p>
  <div
    className="updatedInfo"
    title=""
  >
    Updated 
    Invalid date
  </div>
</article>
`;
